projects/packages/publicize/src/class-social-share-log.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Register the Social Share Log custom post type.
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+/**
+ * Register the Social Share Log custom post type.
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+class Social_Share_Log {
+	const SOCIAL_SHARE_LOG_CPT = 'social_share_log';
+
+	/**
+	 * Initialize the custom post type for social share logs.
+	 */
+	public function init() {
+		self::register_cpt();
+		self::register_share_log_fields();
+	}
+
+	/**
+	 * Register the custom post type for social share logs.
+	 */
+	public function register_cpt() {
+		$args = array(
+			'public'       => false,
+			'show_ui'      => false,
+			'label'        => 'Share Logs',
+			'show_in_rest' => true,
+			'has_archive'  => true,
+			'supports'     => array( 'title', 'custom-fields' ),
+		);
+		register_post_type( self::SOCIAL_SHARE_LOG_CPT, $args );
+	}
+
+	/**
+	 * Registers the custom fields for the social share log.
+	 */
+	public function register_share_log_fields() {
+		$fields = array( 'status', 'message', 'timestamp', 'service', 'connection_id', 'external_name', 'profile_url', 'enabled' );
```

Comment:
Instead of `timestamp` we can make use of post publish date which will help us sort/filter the posts.

Reply:
Also, we will need the `post_id` field that links to the post that was shared.

Reply:
Oh yeah I forgot to update this function for the rest API response. I'll do that. It is passed down from the backend D158508-code

Reply:
We could also make use of `post_content` in place of `message` to have lesser custom fields